sonarqube-ide-visualstudio-roslyn-plugin/src/main/java/org/sonarsource/sonarlint/visualstudio/roslyn/SqvsRoslynSensor.java

```diff
@@ -51,61 +48,33 @@ public SqvsRoslynSensor(HttpAnalysisRequestHandler httpRequestHandler) {
         newIssue
           .forRule(ruleKey)
-          .at(createLocation(newIssue, diag, diagInputFile));
-        handleSecondaryLocations(context, diag, newIssue);
-        handleQuickFixes(context, diag, newIssue);
+          .at(createLocation(newIssue, roslynIssue.getPrimaryLocation(), diagInputFile));
+        handleSecondaryLocations(context, roslynIssue, newIssue);
+        // TODO by https://sonarsource.atlassian.net/browse/SLVS-2492 handle quickfixes once they are returned from the server
+        /* handleQuickFixes(context, roslynIssue, newIssue); */
         newIssue.save();
       }
     }
   }
 
-  private static void handleQuickFixes(SensorContext context, Diagnostic diag, NewIssue newIssue) {
-    var quickFixes = diag.getQuickFixes();
-    if (quickFixes != null && quickFixes.length > 0) {
-      newIssue.setQuickFixAvailable(true);
-      for (var quickFix : quickFixes) {
-        handleQuickFix(context, quickFix, newIssue);
-      }
-    }
-  }
-
-  static void handleQuickFix(SensorContext context, QuickFix quickFix, NewIssue newIssue) {
-    var newQuickFix = newIssue.newQuickFix();
-    newQuickFix.message(quickFix.getMessage());
-    for (Fix fix : quickFix.getFixes()) {
-      var fixInputFile = findInputFile(context, Paths.get(fix.getFilename()));
-      if (fixInputFile != null) {
-        var newInputFileEdit = newQuickFix.newInputFileEdit()
-          .on(fixInputFile);
-        for (QuickFixEdit edit : fix.getEdits()) {
-          var newTextEdit = newInputFileEdit.newTextEdit()
-            .at(fixInputFile.newRange(edit.getStartLine(), edit.getStartColumn() - 1, edit.getEndLine(), edit.getEndColumn() - 1))
-            .withNewText(edit.getNewText());
-          newInputFileEdit.addTextEdit(newTextEdit);
-        }
-        newQuickFix.addInputFileEdit(newInputFileEdit);
-      }
-    }
-    newIssue.addQuickFix(newQuickFix);
-  }
-
-  private static void handleSecondaryLocations(SensorContext context, Diagnostic diag, NewIssue newIssue) {
-    var additionalLocations = diag.getAdditionalLocations();
-    if (additionalLocations != null) {
-      for (var additionalLocation : additionalLocations) {
-        var additionalFilePath = Paths.get(additionalLocation.getFilename());
-        var additionalFilePathInputFile = findInputFile(context, additionalFilePath);
-        if (additionalFilePathInputFile != null) {
-          newIssue.addLocation(createLocation(newIssue, additionalLocation, additionalFilePathInputFile));
+  private static void handleSecondaryLocations(SensorContext context, RoslynIssue diag, NewIssue newIssue) {
+    var flows = diag.getFlows();
+    for (var flow : flows) {
+      for (var flowLocation : flow.getLocations()) {
+        var filePath = Paths.get(flowLocation.getFilePath());
+        var inputFile = findInputFile(context, filePath);
+        if (inputFile != null) {
+          newIssue.addLocation(createLocation(newIssue, flowLocation, inputFile));
```

Comment:
how would this work with multiple flows? we don't have that yet, but im still curious